src/scripts/main.js

```diff
@@ -1,11 +1,20 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(`${eventName}`, () => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
+      );
+    });
+  });
```

Comment:
The `waitFor` function is correctly implemented to return a promise that resolves when the specified event occurs on the given element. However, consider adding error handling in case the element is null or the event listener fails to attach.